lib/jsdom/browser/Window.js

```diff
@@ -154,6 +154,15 @@ function Window(options) {
       lastUrl = fixUrlForBuggyXhr(resolveHref(window.document.URL, url));
       return xhr._open(method, lastUrl, async, user, password);
     };
+    xhr._getAllResponseHeaders = xhr.getAllResponseHeaders;
+    xhr.getAllResponseHeaders = function () {
+      if (window.document.location.protocol === "file:") {
+        // Monkey patch this function for files, the node-xhr module doesn't properly
+        // set the response for file urls.
+        return null;
+      }
+      return xhr.getAllResponseHeaders();
```

Comment:
Infinite recursion. Probably need a test that this works in non-file cases to catch this kind of bug.